dependencies.yaml

```diff
@@ -179,6 +179,7 @@ dependencies:
           - c-compiler
           - cxx-compiler
           - dlpack>=0.5,<0.6.0a0
+          - zlib
```

Comment:
@wence- Can you comment on where/how we use zlib? Does it need to be in the conda package dependencies (meta.yml for libcudf, or cudf, …) as well?

Reply:
Is there an issue documenting the problems you see in builds?

Reply:
We support zlib compression/decompression in ORC reading/writing, so we should be including it in our recipe I think.

Reply:
This is what I was missing: https://github.com/rapidsai/cudf/issues/14021#issuecomment-1701327693